packages/pluggableWidgets/datagrid-dropdown-filter-web/src/DatagridDropdownFilter.tsx

```diff
@@ -87,16 +89,27 @@ function getFilterCondition(listAttribute: ListAttributeValue, values: FilterOpt
         return undefined;
     }
 
-    const filterAttribute = attribute(listAttribute.id);
+    const { id, type } = listAttribute;
+    const filterAttribute = attribute(id);
 
     if (values.length > 1) {
-        return or(...values.map(filter => equals(filterAttribute, literal(filter.value))));
+        return or(...values.map(filter => equals(filterAttribute, literal(checkValue(filter.value, type)))));
     }
 
     const [filterValue] = values;
     if (filterValue.value) {
-        return equals(filterAttribute, literal(filterValue.value));
+        return equals(filterAttribute, literal(checkValue(filterValue.value, type)));
     }
 
     return undefined;
 }
+
+function checkValue(value: string, type: string): boolean | string {
+    if (type === "Boolean") {
+        if (value !== "true" && value !== "false") {
```

Comment:
Are these cases realistic for a value of type `Boolean`? 🤔

Reply:
Users can still add manual value that doesnt match with the expected.. as this is String (Result of the expression attribute)

Reply:
🤷